src/magnet_uri.cpp

```diff
@@ -215,21 +215,23 @@ namespace libtorrent {
 			url = uri.substr(pos, find(uri, "&", pos) - pos);
 		}
 
-		std::string btih = unescape_string(url_has_argument(uri, "xt"), ec);
-		if (ec)
-		{
-			return;
-		}
-		else if (btih.empty())
+		string_view btih = url_has_argument(uri, "xt");
+		std::string unescaped_btih;
+		if (btih.empty())
 		{
 			ec = errors::missing_info_hash_in_uri;
 			return;
 		}
-
-		if (btih.substr(0, 9) != "urn:btih:")
+		else if (btih.substr(0, 9) != "urn:btih:")
```

Comment:
technically, it's possible for part of the hex-encoded info-hash to be escaped as well (redundantly, just like the ":"). So, although I appreciate the attempt to avoid extra work, I think scanning for a "%" may be unavoidable to support this properly